cmd/check-markdown-rs/src/main.rs

```diff
@@ -0,0 +1,568 @@
+use std::collections::HashSet;
+use comrak::arena_tree::Node;
+use comrak::{
+    nodes::{Ast, NodeValue},
+    parse_document, Arena, ComrakOptions,
+};
+use comrak::nodes::LineColumn;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+use url::Url;
+use async_std::task;
+use std::path::{Path, PathBuf};
+use std::cmp::Ord;
+use std::cmp::Ordering;
+
+
+// Add a custom structure to hold the heading information
+#[derive(Debug, PartialEq, Eq, Clone)]
+struct HeadingInfo {
+    level: u32,
+    id: String,
+    text: String,
+}
+
+// Add implementation for PartialOrd and Ord to sort the HeadingInfo
+impl PartialOrd for HeadingInfo {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for HeadingInfo {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.level
+            .cmp(&other.level)
+            .then_with(|| self.text.cmp(&other.text))
+    }
+}
+
+// Main function
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args: Vec<String> = env::args().collect();
+    let arena = Arena::new();
+
+    if args.len() < 2 {
+        eprintln!("Usage: {} <input_file>", args[0]);
+        return Ok(());
```

Comment:
Nit: Ideally, this would return an `Err` as this is it invalid to call the tool without specifying a file.

Reply:
i think this is fixed now